PUBLISH.md

```diff
@@ -0,0 +1,21 @@
+# Instructions for publishing a new SDK version
```

Comment:
by publishing do we mean on GH or on a package distribution system?  worth clarifying as we have learned from ccip-contracts etc that gh tagged versions may not map to NPM release versions etc as there may be github tagged versions that are not "ready" for NPM release.

Please can you update these instructions to distinguish between process for gh tagging vs release tagging.